vms/platformvm/warp/messages/payload.go

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package messages
+
+import (
+	"errors"
+	"fmt"
+)
+
+var errWrongType = errors.New("wrong payload type")
+
+// Payload provides a common interface for all payloads implemented by this
+// package.
+type Payload interface {
+	// Bytes returns the binary representation of this payload.
+	Bytes() []byte
+
+	// initialize the payload with the provided binary representation.
+	initialize(b []byte)
+}
+
+func Parse(bytes []byte) (Payload, error) {
+	var payload Payload
+	if _, err := Codec.Unmarshal(bytes, &payload); err != nil {
+		return nil, err
+	}
+	payload.initialize(bytes)
```

Comment:
This indirection is required if we want to be able to call `Parse` on `Payload` interfaces, rather than concrete types. @darioush would have more context here, but I imagine the final implementation of the [signature request handler](https://github.com/ava-labs/avalanchego/pull/3279) would just call a `Verify` method on the `Payload` that encapsulates the payload-specific checks.

Reply:
Also worth noting that a similar pattern is used to parse blocks: https://github.com/ava-labs/avalanchego/blob/master/vms/proposervm/block/parse.go#L50

Reply:
`Payload` is an interface - which is why we set the bytes like this.